lib/rolodex/utils.ex

```diff
@@ -59,4 +62,36 @@ defmodule Rolodex.Utils do
       _ -> nil
     end
   end
+
+  @doc """
+  Grabs the description and metadata map associated with the given function via
+  `@doc` annotations.
+  """
+  @spec fetch_doc_annotation(module(), atom()) :: {:ok, binary(), map()} | {:error, :not_found}
+  def fetch_doc_annotation(controller, action) do
+    controller
+    |> Code.fetch_docs()
+    |> Tuple.to_list()
+    |> Enum.at(-1)
+    |> Enum.find(fn
+      {{:function, ^action, _arity}, _, _, _, _} -> true
+      _ -> false
+    end)
+    |> case do
+      {_, _, _, desc, metadata} -> {:ok, desc, metadata}
+      _ -> {:error, :not_found}
+    end
+  end
+
+  @doc """
+  Determines if the given module exports the given function
+  """
+  @spec has_function?(module(), atom(), integer()) :: boolean()
+  def has_function?(module, func, arity) do
```

Comment:
Correct me if wrong, but isn't this what `Module.defines?` does?

Reply:
You're not wrong, and TIL :)